components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtImpl.java

```diff
@@ -230,6 +233,51 @@ public void deactivateTenant(String tenantUniqueID) throws TenantMgtException {
         }
     }
 
+    @Override
+    public void deleteTenantMetaData(String tenantUniqueIdentifier) throws TenantMgtException {
+
+        ServerConfigurationService
+                serverConfigurationService = TenantMgtServiceComponent.getServerConfigurationService();
+
+        if (!Boolean.parseBoolean(serverConfigurationService.getFirstProperty("Tenant.TenantDelete"))) {
+            throw new TenantManagementClientException(ERROR_CODE_TENANT_DELETION_NOT_ENABLED.getCode(),
+                    ERROR_CODE_TENANT_DELETION_NOT_ENABLED.getMessage());
+        }
+        TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
+        if (tenantManager != null) {
+            Tenant tenant;
+            try {
+                tenant = tenantManager.getTenant(tenantUniqueIdentifier);
+            } catch (UserStoreException e) {
+                throw new TenantManagementServerException("Error while getting the tenant with the tenant " +
+                        "unique id :" + tenantUniqueIdentifier + " .", e);
+            }
+
+            if (tenant != null) {
```

Comment:
Can't we directly call, deleteTenant(String tenantDomain) method in TenantMgtAdminService

Reply:
No, as discussed offline the way the admin service handles the delete tenant is different.